glean-core/src/metrics/ping.rs

```diff
@@ -208,47 +198,59 @@ impl PingType {
                 false
             }
             Some(ping) => {
-                // This metric is recorded *after* the ping is collected (since
-                // that is the only way to know *if* it will be submitted). The
-                // implication of this is that the count for a metrics ping will
-                // be included in the *next* metrics ping.
-                glean
-                    .additional_metrics
-                    .pings_submitted
-                    .get(ping.name)
-                    .add_sync(glean, 1);
-
-                if let Err(e) = ping_maker.store_ping(glean.get_data_path(), &ping) {
-                    log::warn!("IO error while writing ping to file: {}. Enqueuing upload of what we have in memory.", e);
-                    glean.additional_metrics.io_errors.add_sync(glean, 1);
-                    // `serde_json::to_string` only fails if serialization of the content
-                    // fails or it contains maps with non-string keys.
-                    // However `ping.content` is already a `JsonValue`,
-                    // so both scenarios should be impossible.
-                    let content =
-                        ::serde_json::to_string(&ping.content).expect("ping serialization failed");
-                    // TODO: Shouldn't we consolidate on a single collected Ping representation?
-                    let ping = PingPayload {
-                        document_id: ping.doc_id.to_string(),
-                        upload_path: ping.url_path.to_string(),
-                        json_body: content,
-                        headers: Some(ping.headers),
-                        body_has_info_sections: self.0.include_info_sections,
-                        ping_name: self.0.name.to_string(),
-                    };
-
-                    glean.upload_manager.enqueue_ping(glean, ping);
-                    return true;
-                }
+                if self.enabled(glean) {
```

Comment:
It might be the m-c C++ style guide in my head, but I think I'd prefer this to take the form of an early return rather than a 52-line context.

Reply:
I don't mean completely early, just taking this line and appending `return false; }` after it instead of indenting the following 5 handfuls of lines?

Reply:
Okay, inverted the if to check for the early out to reduce nesting depth.